src/main/kotlin/org/rust/lang/core/types/RustEnumType.kt

```diff
@@ -4,12 +4,21 @@ import com.intellij.codeInsight.completion.CompletionUtil
 import org.rust.lang.core.psi.RustEnumItemElement
 import org.rust.lang.core.types.visitors.RustTypeVisitor
 
-class RustEnumType(enum: RustEnumItemElement) : RustStructOrEnumTypeBase() {
+class RustEnumType(
+    enum: RustEnumItemElement,
+    typeArguments: List<RustType> = emptyList()
+) : RustStructOrEnumTypeBase(typeArguments) {
```

Comment:
Concrete benefits stem from the fact that we use statically-typed language and you suggest to fallback to duck-typing mode: `StructType` in general has nothing to do with substitution and should continue to be doing that way instead of effectively being `StructTypeWithSubstitution` with empty substitution list. Let's not accumulate technical debt.